x-pack/plugins/apm/public/hooks/use_fetch_triggered_alert_history.ts

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AsApiContract } from '@kbn/actions-plugin/common';
+import { HttpSetup } from '@kbn/core/public';
+import {
+  ALERT_DURATION,
+  ALERT_RULE_UUID,
+  ALERT_START,
+  ALERT_STATUS,
+  ALERT_TIME_RANGE,
+} from '@kbn/rule-data-utils';
+import { BASE_RAC_ALERTS_API_PATH } from '@kbn/rule-registry-plugin/common';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+
+interface UseFetchTriggeredAlertsHistoryProps {
+  features: string;
+  ruleId: string;
+}
+interface FetchTriggeredAlertsHistory {
+  totalTriggeredAlerts: number;
+  histogramTriggeredAlerts: Array<{
+    key_as_string: string;
+    key: number;
+    doc_count: number;
+  }>;
+  error?: string;
+  avgTimeToRecoverUS: number;
+}
+
+interface TriggeredAlertsHistory {
+  isLoadingTriggeredAlertHistory: boolean;
+  errorTriggeredAlertHistory?: string;
+  triggeredAlertsData?: FetchTriggeredAlertsHistory;
+}
+export function useFetchTriggeredAlertsHistory({
```

Comment:
It feels like most of this belongs as a client in the alerts plugin. What's specific to APM here?

Reply:
If this is a client, we can import it in apm and use it like:

```
alertsClient.getTriggerdAlertsHistory({ features: 'apm', ruleId })
```

Reply:
👍